soda_machine.rb

```diff
@@ -1,18 +1,39 @@
+require_relative 'soda'
+
 class SodaMachine
   attr_reader :sodas, :cash
+  attr_accessor :current_inventory_count
+  
 
   def initialize(args = {})
     @sodas = args[:sodas]
     @cash = args[:cash]
-  end
-
-  def current_inventory_count
+    @current_inventory_count = @sodas.length
   end
 
   def find_soda(soda_brand)
+    i = 0
+    while i < @sodas.length
+      if @sodas[i].brand == soda_brand 
+        @sodas[i]
+        break
+      end
+      i += 1
+    end
```

Comment:
Can you re-write this using an enumerable (like `.each`, `.map`, `.find`, `.reduce` etc)? Get comfortable with them!